로그인 UI 구현 및 axios instance 로그인 처리 추가

jma1020 opened the pull request:
# Pull Request
- [x]  Feature
## Summary

로그인 UI 구현 및 로그인 처리를 위한 axios instance 설정 추가

## Describe your changes

- 로그인 UI 구현
- 로그인 Link router 연결
- 로그인 처리를 위한 axios instance 설정 추가
- access token 요청 및 저장을 위한 interceptors 코드 추가

## Issue number and link

jma1020 committed:
feat: 로그인 cookie 테스트

- 기존 api axios를 이용하여 login 테스트

jma1020 committed:
Merge branch 'develop' of https://github.com/TRIPONG-2022/frontend into feature/login-test

jma1020 committed:
💄 IconButton xlg 사이즈 추가

jma1020 committed:
⚡️ 에러 체크 및 headers 확인

jma1020 committed:
feat: 로그인 시 받아올 user 정보 redux세팅

- userSlice 생성
- userSlice.reducer store 연결

jma1020 committed:
feat: 로그인 페이지 UI 및 oauth 라우터 연결

- 로그인 페이지 UI 추가
- oauth 링크 라우터 연결

jma1020 committed:
💄 간단한 UI 수정

jma1020 committed:
🔨 https 적용을 위한 설정 추가 및 cross-env, local-ssl-proxy 의존성 추가

jma1020 committed:
🔥 사용하지 않는 코드 삭제

jma1020 committed:
🔨 withCredentials 설정 추가

seongjin2427 commented on pages/auth/login.tsx:
```diff
@@ -16,45 +23,242 @@ const LoginPage: NextPage = () => {
+          <SingUpBtn onClick={() => router.push('/auth/join')}>
+            회원가입
+          </SingUpBtn>
+        </SignUp>
```
앗 갑자기 생각난건데,
저희 비밀번호를 까먹었을 때, 이메일 인증하고 비밀번호 재설정 하는 프로세스가 있었던걸로 기억합니닷!
그래서 비밀번호를 까먹었을 때에 대한 Link도 하나 필요 할 것 같습니다!

jma1020 replied:
아이디 찾기 Link 추가해서 올려놓았습니다!!

jma1020 committed:
⚡️ 아이디찾기 페이지 연결

jma1020 committed:
🔨 login instance 요청 url 수정

jma1020 committed:
🚑 https 적용했던 코드 삭제 및 복구

jma1020 committed:
feat: Authorization 쿠키 저장 및 요청을 axios instance 변경

- instance  withCredentials 설정 추가
- 요청 시 header에 access_token을  보내기 위한 request interceptors
  추가
- 응답 시 갱신된  access_token을 저장하기 위한 response interceptors
  추가
- instance 수정으로 불필요하게된 auth 요청 코드 수정

jma1020 committed:
🔥 instance 수정으로 불필요하게 된 auth 코드 부분 삭제

sangrimlee commented on api/auth.ts:
```diff
@@ -3,21 +3,23 @@ import instance from './instance';
-      loginPwd,
+      password,
     });
     console.log(data);
```
console.log는 지워주시면 좋을 것 같아요 😄

jma1020 replied:
넵!

sangrimlee commented on package.json:
```diff
@@ -19,6 +19,7 @@
     "@hookform/resolvers": "^2.9.1",
     "@reduxjs/toolkit": "^1.8.3",
     "axios": "^0.27.2",
+    "cross-env": "^7.0.3",
```
사용하지 않는 dependency는 지워주면 좋을 것 같아요!

jma1020 replied:
넵!! ㅎㅎ

sangrimlee commented on pages/auth/login.tsx:
```diff
@@ -18,43 +25,223 @@ const LoginPage: NextPage = () => {
 
   const onSubmit = (data: LoginSchema) => {
     console.log(data);
```
console.log도 이제는 차근차근 지워나가보아요 👍🏼

jma1020 replied:
넵!

sangrimlee commented on pages/auth/login.tsx:
```diff
@@ -18,43 +25,223 @@ const LoginPage: NextPage = () => {
+    login(data);
   };
 
+  // 로그인 시 token을 받고,  token있을 경우에
```
주석이 달리다 만 것 같아요!

jma1020 replied:
앗  아예 삭제해버렸습니다 ㅎㅎ

sangrimlee commented on pages/auth/login.tsx:
```diff
@@ -18,43 +25,223 @@ const LoginPage: NextPage = () => {
+  max-width: 32rem;
+  margin: 0 auto;
+  padding: 0 1.25rem;
+  @media (min-width: 768px) {
```
screens에 미리 만들어둔 미디어 쿼리를 사용하면 좋을 것 같아요!

jma1020 replied:
넵!!

sangrimlee commented on pages/auth/login.tsx:
```diff
@@ -18,43 +25,223 @@ const LoginPage: NextPage = () => {
+            {...register('password')}
+          />
+          <ForgotContainer>
+            <ForgotPwdText onClick={() => router.push('/auth/find-id')}>
+              아이디 찾기
+            </ForgotPwdText>
+            <ForgotPwdText
+              pwd
+              onClick={() => router.push('/auth/find-password')}
+            >
+              비밀번호 찾기
+            </ForgotPwdText>
```
이부분은 router를 사용하지말고 next/link를 사용하면 될 것 같은데 router를 사용하신 이유가 있을까요??

jma1020 replied:
앗 기존에 만들었던 oauth에서 router를 사용하고 있어가지고,, 이왕 가져온겸 그냥 router를 사용할까 했었는데 별로일까요 ㅎㅎ?..

jma1020 committed:
refact: localStorage null 값 빈 문자열로 리팩토링

- localStorage.getItem null 값 빈 문자열로 대체

jma1020 committed:
refact: LoginPage 코드 컴포넌트 단위로 분리

- Divider 컴포넌트 생성
- Forgot 컴포넌트 생성
- SignUp 컴포넌트 생성

jma1020 committed:
chore: 사용하지 않는 의존성 제거 및 devDependencies로 이동

- local-ssl-proxy 제거
- cross-env devDependencies로 이동

jma1020 committed:
fix: 불필요한 console 삭제

jma1020 commented on components/Login/index.ts:
```diff
@@ -0,0 +1,3 @@
+export * from './Divider';
+export * from './Forgot';
+export * from './SignUp';
```
해당 부분 저희가 기존에 사용하는 컴포넌트 export 방식으로 사용할 때 LoginPage에서 import가 커지는 것 같아서 ㅠㅠ 이런 형식으로 작성해보았습니다!! 피드백 부탁드립니다 ㅎㅎ..

jma1020 committed:
fix: 불필요한 코드 삭제

jma1020 committed:
refact: oauth 리팩토링

- router Link로 변경
- OAUTH DATA 생성 및 적용

jma1020 committed:
refact: Login 컴포넌트 리팩토링

- 혼용될 수 있는 변수명 수정
- Link 프리페치를 위한 router -> Link 로 변경
- css tag 시멘틱하게 수정

jma1020 committed:
fix: 사용하지 않는 파일 삭제

seongjin2427 reviewed (approved):
고생하셨습니다!! 😀
